pkg/config/config.go

```diff
@@ -40,14 +40,30 @@ func (c *Config) UnmarshalYAML(data []byte) error {
 }
 
 type Provider struct {
-	Name              string `yaml:"name"`
-	ApiKey            string `yaml:"apiKey"`
-	ApiURL            string `yaml:"apiURL"`
-	AlertContacts     string `yaml:"alertContacts"`
-	AlertIntegrations string `yaml:"alertIntegrations"`
-	Username          string `yaml:"username"`
-	Password          string `yaml:"password"`
-	AccountEmail      string `yaml:"accountEmail"`
+	Name              string      `yaml:"name"`
+	ApiKey            string      `yaml:"apiKey"`
+	ApiURL            string      `yaml:"apiURL"`
+	AlertContacts     string      `yaml:"alertContacts"`
+	AlertIntegrations string      `yaml:"alertIntegrations"`
+	Username          string      `yaml:"username"`
+	Password          string      `yaml:"password"`
+	AccountEmail      string      `yaml:"accountEmail"`
+	AppInsightsConfig AppInsights `yaml:"appInsightsConfig"`
```

Comment:
This will get out of hands when we support more and more providers. We might have to use a mapped interface for providerConfig so that we can separate provider specific configuration

Reply:
@waseem-h Thanks, I am not sure if I get your point correctly. But in my opinion, having dedicated struct as a provider-specific config will make further contributions clean and easier.